### ISSUE TITLE
Problem with Event-Handler

### ISSUE THREAD

**Issue by AlexanderBurdenko** [liatrio/terraform-azure-github-runner#108]
Image version: Latest
Image: ghcr.io/liatrio/github-webhook-event-handler:latest

**Describe the bug**
Problem with Event-Handler he doesn't write message to queue.

After running Workflow, i received in Github App webhook payload and send it to Azure function App 
Valid webhook message received. Queued [https://api.github.com/repos/balalallal/1346789] for processing

After this it must be in queue sbq-github-webhook-events but there is no ANY Messages

And inside Runner-Controller i have an error
2024-02-01T12:43:36.156500288Z {"level":30,"time":1706791416140,"pid":1,"hostname":"6caa36257737","msg":"Queued Event Received"}
2024-02-01T12:43:36.548939285Z {"level":30,"time":1706791416544,"pid":1,"hostname":"6caa36257737","msg":"[EventQueue] Process Message: "}

And nothing happened

What can be the problem ?

FYI: in your code i see AzureCredentinal, a question how fucntion can get it ?
Thank you

**Comment by mitchell-liatrio** [comment]
Hey @AlexanderBurdenko, looks like you're running into a good chunk of issues on this implementation.  Mind if I get an email address to invite you into a Slack channel to 🍐 up on this.  

There are likely some areas we need to improve the docs based on the last few issues being raised recently.

**Comment by AlexanderBurdenko** [comment]
> > But for me important to understand what need to do now.
> > Maybe I something miss in configuration? Because I see that nobody wrote about this problem before.
> 
> I'm not 100% sure what the issue is that you're facing which is why I wanted to spin up a chat where we can work through it together and hop on a huddle if needed. My guess is there is something missing in our documentation but from a time management perspective, it would be a lot easier to help resolve the issue you're facing versus trying to replicate it on my end.
> 
> Feel free to message me at [mitchellp@liatrio.com](mailto:mitchellp@liatrio.com) and I can send you an invite to a slack channel

Well, i can provide logs which Function App in showing in Deployment - Logs or in Console where Code + Test in Azure if this help too

**Comment by AlexanderBurdenko** [comment]
@mitchell-liatrio Hello
I have collected all the information which may help to understand why the function can’t write in the queue:

General issue:
Problems with logs
In Function:

Example:
If I will post an incorrect body I receive an error

MS_FUNCTION_AZURE_MONITOR_EVENT 3,.azurewebsites.net,Microsoft.Web/sites/functions/log,FunctionAppLogs,"{'appName':'func-github-webhook-event-handler-gitrunas','roleInstance':'c12312414124','message':'Lacking body, body.action, or body.workflow_job','category':'Function.eventHandler.User','hostVersion':'4.23.1.1','functionInvocationId’:’44324324’,’functionName':'eventHandler','hostInstanceId’:’242424234’,’level':'Warning','levelId':3,'processId':1}",02/02/2024 10:41:20
2024-02-02T10:41:20.290424197Z [40m[1m[33mwarn[39m[22m[49m: Function.eventHandler.User[0]
2024-02-02T10:41:20.290446697Z       Lacking body, body.action, or body.workflow_job

But sometimes if I put body with correct information it’s give me error in Github but in Function I don’t have any errors
Only that FUNCTION correctly executed

Why logs don’t show any other errors if it happened?
FYI: I’m watching logs 

Problem with writing in queue:

Steps which I did
**First** - Running Workflow with correct parameters

**Second** - Checked that payload is correct and Response of Function has next view:

**Status:** 200
**Body Response:**
Valid webhook message received. Queued [https://api.github.com/repos/BLABLA/blalal/actions/runs/123567] for processing

In Function logs in the console of Azure, I see next logs:

MS_FUNCTION_AZURE_MONITOR_EVENT 4,func-github-webhook-event-handler-gitrunas.azurewebsites.net,Microsoft.Web/sites/functions/log,FunctionAppLogs,"{'appName':'func-github-webhook-event-handler-gitrunas','roleInstance':'c12312414124','message':'Executing Functions.eventHandler (Reason=This function was programmatically called via the host APIs., Id=c12312414124)','category':'Function.eventHandler','hostVersion':'4.23.1.1','functionInvocationId':'c12312414124','functionName':'Functions.eventHandler','hostInstanceId':'c12312414124','level':'Information','levelId':2,'processId':1,'eventId':1,'eventName':'FunctionStarted'}",02/01/2024 17:48:06
2024-02-01T17:48:06.345691988Z info: Function.eventHandler[1]
2024-02-01T17:48:06.345699988Z       Executing 'Functions.eventHandler' (Reason='This function was programmatically called via the host APIs.', Id=c12312414124)
2024-02-01T17:48:06.554677518Z info: Function.eventHandler[2]
2024-02-01T17:48:06.554783319Z       Executed 'Functions.eventHandler' (Succeeded, Id=c12312414124, Duration=210ms)


I saw in your code https://github.com/liatrio/terraform-azure-github-runner/blob/main/app/event-handler/eventHandler/index.js that must be the next step:

 const sender = await getWebHookEventsQueueSender(context);

        await sender.sendMessages({
            body: req.body,
        });
        context.log.verbose("Placed message on queue", sender);

But I don’t see that this step doing, because i don't see message in my queue 

**Name Status Message count Active messages Dead-letter messages Scheduled messages Max size Enable partitioning**
sbq-github-webhook-events Active 0 0 0 0 5120 MB false

In Runner-Controller happened strange:

My logs inside Web-App:
2024-02-01T12:43:36.156500288Z {"level":30,"time":1706791416140,"pid":1,"hostname":"6caa36257737","msg":"Queued Event Received"}
2024-02-01T12:43:36.548939285Z {"level":30,"time":1706791416544,"pid":1,"hostname":"6caa36257737","msg":"[EventQueue] Process Message: "}

And Runner-Contoller write to queue message
**Name Status Message count Active messages Dead-letter messages Scheduled messages Max size Enable partitioning**
sbq-github-runners Active 1 1 0 0 5120 MB false

But VM doesn’t create.

That’s all the information that I collected. Hope this helps
